pallets/governance/src/application.rs

```diff
@@ -108,30 +142,29 @@ pub fn submit_application<T: crate::Config>(
     Ok(())
 }
 
-pub fn accept_application<T: crate::Config>(application_id: u32) -> DispatchResult {
-    let application = crate::AgentApplications::<T>::get(application_id)
-        .ok_or(crate::Error::<T>::ApplicationNotFound)?;
-
-    match application.action {
-        ApplicationAction::Add => {
-            crate::Whitelist::<T>::insert(application.agent_key.clone(), ());
-            crate::Pallet::<T>::deposit_event(crate::Event::<T>::WhitelistAdded(
-                application.agent_key,
-            ));
-        }
-        ApplicationAction::Remove => {
-            crate::Whitelist::<T>::remove(&application.agent_key);
-            crate::Pallet::<T>::deposit_event(crate::Event::<T>::WhitelistRemoved(
-                application.agent_key,
-            ));
+pub fn accept_application<T: crate::Config>(
+    resolver: T::AccountId,
+    application_id: u32,
+) -> DispatchResult {
+    let application = crate::AgentApplications::<T>::mutate(application_id, |app| {
+        if let Some(app) = app {
```

Comment:
Just realized we are not checking if it's Open. Maybe we should move to a `.accept()` on the impl, like `.revoke()`, doing the check.